sources/osg/Matrix.js

```diff
@@ -651,12 +692,16 @@ define( [
             var tmpVec = [ 0.0, 0.0, 0.0 ];
 
             return function ( matrix, vector, result ) {
-                var d = 1.0 / ( matrix[ 3 ] * vector[ 0 ] + matrix[ 7 ] * vector[ 1 ] + matrix[ 11 ] * vector[ 2 ] + matrix[ 15 ] );
-
                 if ( result === undefined ) {
-                    Notify.warn( 'no matrix destination !' );
-                    result = Matrix.create();
+                    Notify.warn( 'no vec3 destination !' );
+                    result = Vec3.create();
                 }
+                if ( Matrix.isIdentity( matrix ) ) {
```

Comment:
could you bench this, not sure statistically it's better to test if a matrix is identity before doing the operation. I mean in js.